### PR TITLE
[bitnami/kafka] add schedulerName for all resources

### PR DESCRIPTION
**Pull request by yevgeny-z**
**Description of the change**
Add `schedulerName` option for all kafka chart objects

**Benefits**
Allow configure `schedulerName` for all pods created by kafka chart 

**Possible drawbacks**
none

**Applicable issues**
none

**Additional information**
none

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/kafka/templates/kafka-metrics-deployment.yaml [hunk 1]
```diff
@@ -26,7 +26,10 @@ spec:
       labels: {{- include "common.labels.standard" . | nindent 8 }}
         app.kubernetes.io/component: metrics
     spec:
-{{- include "kafka.imagePullSecrets" . | indent 6 }}
+      {{- include "kafka.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.metrics.kafka.schedulerName }}
+      schedulerName: {{ .Values.metrics.kafka.schedulerName }}
```
Comment (andresbono) [inline bitnami/kafka/templates/kafka-metrics-deployment.yaml:31]: Should this filed be quoted for safety?
```suggestion
      schedulerName: {{ .Values.metrics.kafka.schedulerName | quote }}
```

File: bitnami/kafka/templates/kafka-provisioning.yaml [hunk 2]
```diff
@@ -24,7 +24,10 @@ spec:
         {{- end }}
       annotations:
     spec:
-{{- include "kafka.imagePullSecrets" . | indent 6 }}
+      {{- include "kafka.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.provisioning.schedulerName }}
+      schedulerName: {{ .Values.provisioning.schedulerName }}
```
Comment (andresbono) [inline bitnami/kafka/templates/kafka-provisioning.yaml:29]: Should this filed be quoted for safety?
```suggestion
      schedulerName: {{ .Values.provisioning.schedulerName | quote }}
```

File: bitnami/kafka/templates/statefulset.yaml [hunk 3]
```diff
@@ -65,6 +65,9 @@ spec:
       {{- if .Values.hostAliases }}
       hostAliases: {{- include "common.tplvalues.render" (dict "value" .Values.hostAliases "context" $) | nindent 8 }}
       {{- end }}
+      {{- if .Values.schedulerName }}
+      schedulerName: {{ .Values.schedulerName }}
```
Comment (andresbono) [inline bitnami/kafka/templates/statefulset.yaml:69]: Should this filed be quoted for safety?
```suggestion
      schedulerName: {{ .Values.schedulerName | quote }}
```

**Review by andresbono — COMMENTED** [review]
Thanks for your contribution! Could you please check my suggestions?

**Comment by yevgeny-z** [comment]
@andresbono in helm stable repo usually don't use filter `quote` for this variable. In bitnami repo 50/50 with or with out filter.

What your practice about it?

**Comment by andresbono** [comment]
@yevgeny-z, I'm suggesting the `quote` filter because we have faced issues when some object names are numbers, then the types don't match. The [type for `schedulerName` is `string`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podspec-v1-core) so we can say that quotes are safe to add.

File: bitnami/kafka/templates/kafka-metrics-deployment.yaml (hunk 1 above)
Comment (yevgeny-z) [reply bitnami/kafka/templates/kafka-metrics-deployment.yaml]: applied

File: bitnami/kafka/templates/kafka-provisioning.yaml (hunk 2 above)
Comment (yevgeny-z) [reply bitnami/kafka/templates/kafka-provisioning.yaml]: applied

File: bitnami/kafka/templates/statefulset.yaml (hunk 3 above)
Comment (yevgeny-z) [reply bitnami/kafka/templates/statefulset.yaml]: applied

**Review by andresbono — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**